### PR TITLE
Report endpoints

### PR DESCRIPTION
**Pull request by Luuxo77**
- Simplify Report class
- Add report endpoints
- Add more ReportService tests
- Add new exceptions
- Add global string to enum converter

### CODE REVIEW

File: Api/Controllers/ReportsController.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    [HttpDelete("{id}")]
+    [SwaggerOperation("Delete report")]
+    public ActionResult DeleteReport([FromRoute] Guid id)
+    {
+        reportService
+            .AsUser(User.GetGuid())
+            .DeleteReport(id);
+
+        return NoContent();
```
Comment (majewska-mi) [inline Api/Controllers/ReportsController.cs]: In other DELETE endpoints we always return Ok when successful. I think we should stay consistent

File: Domain/Services/Implementations/ReportService.cs [hunk 2]
```diff
@@ -24,38 +25,63 @@ public ReportType GetByGuid<ReportType>(Guid guid)
...
-        string details, ReportCategory category)
-        where TargetType : BaseEntity
-        where ReportType : Report
+    public Report GetByGuid(Guid guid)
+        => GetByGuid<Report>(guid);
+
+    public IEnumerable<Report> GetReports(ICollection<ReportType> types, bool open, bool closed)
     {
-        AllowOnlyLoggedIn();
+        AllowOnlyUser(ActingUser);
```
Comment (majewska-mi) [inline Domain/Services/Implementations/ReportService.cs]: This changes nothing. AllowOnlyLoggedIn enforces ActingUser not to be null. AllowOnlyUser additionally checks whether ActingUser is equal to passed in User (here also ActingUser), so results are same.

**Review by majewska-mi — COMMENTED** [review]
I left 2 comments

File: Domain/Services/Implementations/ReportService.cs (hunk 2 above)
Comment (Luuxo77) [reply Domain/Services/Implementations/ReportService.cs]: You are right, that doesn't make much sense to check if object is equal to itself.

**Review by majewska-mi — APPROVED** [review]
I noticed, that when calling GetReports with empty Types parameter (happens when you choose "--" in Swagger) an "Invalid value" is returned. I don't know if that's intended. Other than that LGTM

**Comment by Luuxo77** [comment]
In that case it is intended as sending empty array would mean getting no reports. For now I will merge this, but if it will be needed I will change it.

**Pull request merged**